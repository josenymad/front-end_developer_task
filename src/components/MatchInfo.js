import React from "react";
import arsenalBadge from "../images/arsenal.png";
import fulhamBadge from "../images/fulham.png";
import DetailedMatchStats from "./DetailedMatchStats";
import "../styles/match-info.css";

const MatchInfo = ({ matchData }) => {
  // destructuring to save variables for rendering
  const { contestant, liveData } = matchData;
  const homeTeamName = contestant[0].name;
  const awayTeamName = contestant[1].name;
  const { home: homeTeamScore, away: awayTeamScore } =
    liveData.matchDetails.scores.total;

  return (
    <div className="match-info">
      <div className="match-info__results">
        <div className="match-info__home-team">
          <img
            src={arsenalBadge}
            alt="Home Team Badge"
            className="match-info__home-team__badge"
          />
          <h2 className="match-info__home-team__name">{homeTeamName}</h2>
          <h3 className="match-info__home-team__score">{homeTeamScore}</h3>
        </div>
        <div className="match-info__away-team">
          <img
            src={fulhamBadge}
            alt="Away Team Badge"
            className="match-info__away-team__badge"
          />
          <h2 className="match-info__away-team__name">{awayTeamName}</h2>
          <h3 className="match-info__away-team__score">{awayTeamScore}</h3>
        </div>
      </div>

      <DetailedMatchStats liveData={liveData} />
    </div>
  );
};

export default MatchInfo;
