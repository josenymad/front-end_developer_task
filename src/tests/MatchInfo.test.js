import React from "react";
import { render, screen } from "@testing-library/react";
import MatchInfo from "../components/MatchInfo";

describe("Match Info", () => {
  const validProps = {
    home: "Arsenal",
    away: "Fulham",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<MatchInfo teams={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Arsenal", () => {
    render(<MatchInfo teams={validProps} />);
    const arsenal = screen.getByText(/arsenal/i);

    expect(arsenal).toBeInTheDocument();
  });

  it("renders Fulham", () => {
    render(<MatchInfo teams={validProps} />);
    const fulham = screen.getByText(/fulham/i);

    expect(fulham).toBeInTheDocument();
  });
});
