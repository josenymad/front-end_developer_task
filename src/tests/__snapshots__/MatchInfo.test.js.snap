// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Info renders correctly 1`] = `
<DocumentFragment>
  <div
    class="match-info"
  >
    <div
      class="match-info__results"
    >
      <div
        class="match-info__home-team"
      >
        <img
          alt="Home Team Badge"
          class="match-info__home-team__badge"
          src="arsenal.png"
        />
        <h2
          class="match-info__home-team__name"
        >
          Arsenal
        </h2>
        <h3
          class="match-info__home-team__score"
        >
          2
        </h3>
      </div>
      <div
        class="match-info__away-team"
      >
        <img
          alt="Away Team Badge"
          class="match-info__away-team__badge"
          src="fulham.png"
        />
        <h2
          class="match-info__away-team__name"
        >
          Fulham
        </h2>
        <h3
          class="match-info__away-team__score"
        >
          2
        </h3>
      </div>
    </div>
    <div
      class="detailed-match-stats"
    >
      <h4
        class="detailed-match-stats__full-time__header"
      >
        Full Time
      </h4>
      <div
        class="detailed-match-stats__full-time__container"
      >
        <div
          class="detailed-match-stats__full-time__home-team"
        >
          <p
            class="detailed-match-stats__full-time__home-team__shots"
          >
            19
          </p>
          <p
            class="detailed-match-stats__full-time__home-team__shots-on-target"
          >
            11
          </p>
          <p
            class="detailed-match-stats__full-time__home-team__corners"
          >
            8
          </p>
          <p
            class="detailed-match-stats__full-time__home-team__possession"
          >
            71.1%
          </p>
        </div>
        <div
          class="detailed-match-stats__full-time__descriptions"
        >
          <p
            class="detailed-match-stats__full-time__descriptions__shots"
          >
            Shots
          </p>
          <p
            class="detailed-match-stats__full-time__descriptions__shots-on-target"
          >
            Shots On Target
          </p>
          <p
            class="detailed-match-stats__full-time__descriptions__corners"
          >
            Corners
          </p>
          <p
            class="detailed-match-stats__full-time__descriptions__possession"
          >
            Possession
          </p>
          <p
            class="detailed-match-stats__full-time__descriptions__yellow-cards"
          >
            Yellow Cards
          </p>
          <p
            class="detailed-match-stats__full-time__descriptions__red-cards"
          >
            Red Cards
          </p>
        </div>
        <div
          class="detailed-match-stats__full-time__away-team"
        >
          <p
            class="detailed-match-stats__full-time__away-team__shots"
          >
            8
          </p>
          <p
            class="detailed-match-stats__full-time__away-team__shots-on-target"
          >
            3
          </p>
          <p
            class="detailed-match-stats__full-time__away-team__corners"
          >
            3
          </p>
          <p
            class="detailed-match-stats__full-time__away-team__possession"
          >
            28.9%
          </p>
          <p
            class="detailed-match-stats__full-time__away-team__yellow-cards"
          >
            4
          </p>
          <p
            class="detailed-match-stats__full-time__away-team__red-cards"
          >
            1
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
